File: worker/internal/models/weather.go
package models

type HourlyForecast struct {
	Times                   []string  `json:"times"`
	PrecipitationProbability []float64 `json:"precipitation_probability"`
	Precipitation           []float64 `json:"precipitation"`
}

type WeatherData struct {
	City				 			string  `json:"city"`
	State              				string  `json:"state"`
	Country              			string  `json:"country"`
	Timestamp            			string  `json:"timestamp"`
	TemperatureCelsius   			float64 `json:"temperature_celsius"`
	HumidityPercent    	 			float64 `json:"humidity_percent"`
	WindSpeedMS          			float64 `json:"wind_speed_ms"`
	WindDirectionDegrees 			float64 `json:"wind_direction_degrees"`
	WeatherCode          			int     `json:"weather_code"`
	ApparentTemperature				float64 `json:"apparent_temperature"`
	PrecipitationProbabilityPercent float64 `json:"precipitation_probability_percent"`
	HourlyForecast                  HourlyForecast `json:"hourly_forecast"`
}

File: frontend/weather-app/src/interfaces/Weather.ts
export interface HourlyForecast {
  times: string[];
  precipitation_probability: number[];
  precipitation: number[];
}

export interface WeatherData {
  _id: number;
  city: string;
  state: string;
  country: string;
  timestamp: Date;
  temperature_celsius: number;
  apparent_temperature: number;
  humidity_percent: number;
  wind_speed_ms: number;
  wind_direction_degrees: number;
  weather_code: number;
  precipitation_probability_percent: number;
  hourly_forecast: HourlyForecast; // New field
}
